compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentImplWriter.scala

```diff
@@ -0,0 +1,161 @@
+package fpp.compiler.codegen
+
+import fpp.compiler.analysis._
+import fpp.compiler.ast._
+import fpp.compiler.codegen._
+import fpp.compiler.util._
+
+/** Writes out C++ for component implementation templates */
+case class ComponentImplWriter(
+  s: CppWriterState,
+  aNode: Ast.Annotated[AstNode[Ast.DefComponent]]
+) extends ComponentCppWriterUtils(s, aNode) {
+
+  private val fileName = ComputeCppFiles.FileNames.getComponentImpl(name)
+
+  private val implClassName = name
+
+  private val namespaceIdentList = s.getNamespaceIdentList(symbol)
+
+  def write: CppDoc = {
+    val includeGuard = s.includeGuardFromQualifiedName(symbol, fileName)
+    CppWriter.createCppDoc(
+      s"$name component implementation class",
+      fileName,
+      includeGuard,
+      getMembers,
+      s.toolName,
+      "hpp-template",
```

Comment:
This is what F Prime currently uses for the template suffix, but I think it would be better to use .template.{hpp,cpp}. That way when you open the file you get the syntax highlighting. Sometimes it’s useful to open the template file to see what’s in it, without first copying it over to a file with a correct suffix. @LeStarch does this sound OK?

Reply:
Also I notice that even in this PR, GitHub doesn't highlight the `.hpp-template` reference files as C++. So the files are harder to read than the could be.

Reply:
Yeah, I noticed that too. I think this is a good idea.